Description: Improve the invocation of the pager.
 Also check for the PAGER environment variable.
 Add the -f and -r arguments if the user-specified pager is "less", too.
Forwarded: not-yet
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2016-11-17

--- a/gitless/cli/helpers.py
+++ b/gitless/cli/helpers.py
@@ -73,11 +73,11 @@
     pager = repo.config['core.pager']
   except KeyError:
     pass
-  if pager:
-    cmd = shlex.split(pager)
-    cmd.append(fp)
-  else:
-    cmd = ['less', '-r', '-f', fp]
+  pager = pager or os.environ.get('PAGER', None) or 'less'
+  cmd = shlex.split(pager)
+  if os.path.basename(cmd[0]) == 'less':
+    cmd.extend(['-r', '-f'])
+  cmd.append(fp)
   subprocess.call(cmd, stdin=sys.stdin, stdout=sys.stdout)
 
 
